### DISCUSSION TITLE
why doesn't it work out of the box? unknown errors

### DISCUSSION THREAD

**Discussion by TheRealKraytonian** [LazyVim/LazyVim#4171] [Q&A]
Below I dropped my checkhealth I dont know what is going on and am new to neovim I will also provide images of the errors

![Screenshot 2024-07-24 173619](https://github.com/user-attachments/assets/3d730a79-ae2f-486e-b124-d99a281f2971)
![image](https://github.com/user-attachments/assets/6540bc3b-af44-404c-a5f3-da280de75355)
`
...neovim/nvim-win64/share/nvim/runtime/lua/vim/lsp/rpc.lua:389: attempt to index local 'decoded' (a nil value)
stack traceback:
	...neovim/nvim-win64/share/nvim/runtime/lua/vim/lsp/rpc.lua:389: in function 'handle_body'
	...neovim/nvim-win64/share/nvim/runtime/lua/vim/lsp/rpc.lua:739: in function 'handle_body'
	...neovim/nvim-win64/share/nvim/runtime/lua/vim/lsp/rpc.lua:263: in function <...neovim/nvim-win64/share/nvim/runtime/lua/vim/lsp/rpc.lua:247>
	[C]: in function 'nvim_buf_set_option'
	...data/lazy/nvim-notify/lua/notify/service/buffer/init.lua:103: in function 'render'
	...l/nvim-data/lazy/nvim-notify/lua/notify/service/init.lua:53: in function 'push'
	...Local/nvim-data/lazy/nvim-notify/lua/notify/instance.lua:77: in function <...Local/nvim-data/lazy/nvim-notify/lua/notify/instance.lua:41>
	...m-data/lazy/noice.nvim/lua/noice/view/backend/notify.lua:169: in function '_notify'
	...m-data/lazy/noice.nvim/lua/noice/view/backend/notify.lua:199: in function 'show'
	.../Local/nvim-data/lazy/noice.nvim/lua/noice/view/init.lua:156: in function <.../Local/nvim-data/lazy/noice.nvim/lua/noice/view/init.lua:154>
	[C]: in function 'xpcall'
	.../Local/nvim-data/lazy/noice.nvim/lua/noice/util/call.lua:149: in function <.../Local/nvim-data/lazy/noice.nvim/lua/noice/util/call.lua:134>
	.../Local/nvim-data/lazy/noice.nvim/lua/noice/view/init.lua:163: in function 'display'
	...l/nvim-data/lazy/noice.nvim/lua/noice/message/router.lua:227: in function <...l/nvim-data/lazy/noice.nvim/lua/noice/message/router.lua:160>
	[C]: in function 'xpcall'
	.../Local/nvim-data/lazy/noice.nvim/lua/noice/util/call.lua:149: in function <.../Local/nvim-data/lazy/noice.nvim/lua/noice/util/call.lua:134>
	[C]: in function 'pcall'
	.../Local/nvim-data/lazy/noice.nvim/lua/noice/util/init.lua:146: in function 'fn'
	vim/_editor.lua:477: in function 'cb'
	vim/_editor.lua:263: in function <vim/_editor.lua:262>
`

`

==============================================================================
lazy: require("lazy.health").check()

lazy.nvim ~
- {lazy.nvim} version `11.13.1`
- OK {git} `version 2.43.0.windows.1`
- OK no existing packages found by other package managers
- OK packer_compiled.lua not found

luarocks ~
- checking `hererocks` installation
- OK no plugins require `luarocks`, so you can ignore any warnings below
- OK {python} `Python 3.12.1`
- ERROR {C:/Users/billionaire/AppData/Local/nvim-data/lazy-rocks/hererocks/bin/luarocks} not installed
- WARNING {C:/Users/billionaire/AppData/Local/nvim-data/lazy-rocks/hererocks/bin/lua} version `5.1` not installed
- WARNING Lazy won't be able to install plugins that require `luarocks`.
  Here's what you can do:
  - fix your `luarocks` installation
  - disable *hererocks* with `opts.rocks.hererocks = false`
  - disable `luarocks` support completely with `opts.rocks.enabled = false`

==============================================================================
lazyvim: require("lazyvim.health").check()

LazyVim ~
- OK Using Neovim >= 0.9.0
- WARNING Use Neovim >= 0.10.0 for the best experience
- OK `git` is installed
- OK `rg` is installed
- OK `fd` is installed
- OK `lazygit` is installed

==============================================================================
mason: require("mason.health").check()

mason.nvim ~
- OK mason.nvim version v1.10.0
- OK PATH: prepend
- OK Providers: 
  mason.providers.registry-api
  mason.providers.client
- OK neovim version >= 0.7.0

mason.nvim [Registries] ~
- OK Registry `github.com/mason-org/mason-registry version: 2024-07-24-magical-humor` is installed.

mason.nvim [Core utils] ~
- OK unzip: `UnZip 6.00 of 20 April 2009, by Info-ZIP.  Maintained by C. Spieler.  Send
`
- OK wget: `GNU Wget 1.21.4 built on mingw32.
`
- OK curl: `curl 8.7.1 (Windows) libcurl/8.7.1 Schannel zlib/1.3 WinIDN
`
- OK gzip: `gzip 1.3.12
`
- OK tar: `bsdtar 3.5.2 - libarchive 3.5.2 zlib/1.2.5.f-ipp 
`
- OK pwsh: `7.4.3 Microsoft Windows 10.0.19045 Win32NT
`
- WARNING 7z: not available
  - ADVICE:
    - spawn: 7z failed with exit code - and signal -. 7z is not executable

mason.nvim [Languages] ~
- WARNING Go: not available
  - ADVICE:
    - spawn: go failed with exit code - and signal -. go is not executable
- WARNING luarocks: not available
  - ADVICE:
    - spawn: luarocks failed with exit code - and signal -. luarocks is not executable
- WARNING Ruby: not available
  - ADVICE:
    - spawn: ruby failed with exit code - and signal -. ruby is not executable
- WARNING RubyGem: not available
  - ADVICE:
    - spawn: gem.cmd failed with exit code - and signal -. gem.cmd is not executable
- WARNING Composer: not available
  - ADVICE:
    - spawn: composer.bat failed with exit code - and signal -. composer.bat is not executable
- WARNING PHP: not available
  - ADVICE:
    - spawn: php failed with exit code - and signal -. php is not executable
- OK node: `v22.2.0
`
- WARNING julia: not available
  - ADVICE:
    - spawn: julia failed with exit code - and signal -. julia is not executable
- OK python: `Python 3.12.1
`
- OK java: `openjdk version "1.8.0_412"
`
- OK JAVA_HOME: `openjdk version "1.8.0_412"
`
- OK javac: `Ok`
- OK npm: `10.7.0`
- OK cargo: `cargo 1.79.0 (ffa9cf99a 2024-06-03)`
- OK pip: `pip 24.0 from C:\Python312\Lib\site-packages\pip (python 3.12)

`
- OK python venv: `Ok`

mason.nvim [GitHub] ~
- OK GitHub API rate limit. Used: 0. Remaining: 5000. Limit: 5000. Reset: 7/24/2024 6:29:23 PM.

==============================================================================
noice: require("noice.health").check()

noice.nvim ~
- OK **Neovim** >= 0.9.0
- WARNING **Neovim** >= 0.10 is highly recommended, since it fixes some issues related to `vim.ui_attach`
- OK You're using a GUI that should work ok
- OK **vim.go.lazyredraw** is not enabled
- OK **nvim-notify** is installed
- OK **TreeSitter vim** parser is installed
- OK **TreeSitter regex** parser is installed
- OK **TreeSitter lua** parser is installed
- OK **TreeSitter bash** parser is installed
- OK **TreeSitter markdown** parser is installed
- OK **TreeSitter markdown_inline** parser is installed
- OK `vim.notify` is set to **Noice**
- OK `vim.lsp.handlers["textDocument/hover"]` is set to **Noice**
- OK `vim.lsp.handlers["textDocument/signatureHelp"]` is set to **Noice**
- OK `vim.lsp.handlers["window/showMessage"]` is set to **Noice**
- OK `vim.lsp.util.convert_input_to_markdown_lines` is set to **Noice**
- OK `vim.lsp.util.stylize_markdown` is set to **Noice**

==============================================================================
nvim: require("nvim.health").check()

Configuration ~
- OK no issues found

Runtime ~
- OK $VIMRUNTIME: C:\tools\neovim\nvim-win64\share\nvim\runtime

Performance ~
- OK Build type: RelWithDebInfo

Remote Plugins ~
- OK Up to date

==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `node` found v22.2.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `gcc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: gcc (x86_64-posix-seh-rev0, Built by MinGW-Builds project) 13.2.0
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "10.0.19045",
  sysname = "Windows_NT",
  version = "Windows 10 Pro"
} ~

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - c_sharp             ✓ ✓ ✓ . ✓
  - diff                ✓ . . . ✓
  - graphql             ✓ . . ✓ ✓
  - hcl                 ✓ . ✓ ✓ ✓
  - html                ✓ ✓ ✓ ✓ ✓
  - http                ✓ . . . ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jsdoc               ✓ . . . .
  - json                ✓ ✓ ✓ ✓ .
  - jsonc               ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - luadoc              ✓ . . . .
  - luap                ✓ . . . .
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - printf              ✓ . . . .
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - regex               ✓ . . . .
  - sql                 ✓ . . ✓ ✓
  - terraform           ✓ . ✓ ✓ ✓
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - xml                 ✓ ✓ ✓ ✓ ✓
  - yaml                ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

==============================================================================
provider: health#provider#check

Clipboard (optional) ~
- OK Clipboard tool found: win32yank

Python 3 provider (optional) ~
- WARNING No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
- WARNING Could not load Python 3:
  python3 not found in search path or not executable.
  C:\ProgramData\chocolatey\bin\python3.12.EXE does not have the "neovim" module.
  python3.11 not found in search path or not executable.
  python3.10 not found in search path or not executable.
  python3.9 not found in search path or not executable.
  python3.8 not found in search path or not executable.
  python3.7 not found in search path or not executable.
  C:\Python312\python.EXE does not have the "neovim" module.
  - ADVICE:
    - See :help |provider-python| for more information.
    - You may disable this provider (and warning) by adding `let g:loaded_python3_provider = 0` to your init.vim
- Executable: Not found

Python virtualenv ~
- OK no $VIRTUAL_ENV

Ruby provider (optional) ~
- WARNING `ruby` and `gem` must be in $PATH.
  - ADVICE:
    - Install Ruby and verify that `ruby` and `gem` commands work.

Node.js provider (optional) ~
- Node.js: v22.2.0
- WARNING Missing "neovim" npm (or yarn, pnpm) package.
  - ADVICE:
    - Run in shell: npm install -g neovim
    - Run in shell (if you use yarn): yarn global add neovim
    - Run in shell (if you use pnpm): pnpm install -g neovim
    - You may disable this provider (and warning) by adding `let g:loaded_node_provider = 0` to your init.vim

Perl provider (optional) ~
- WARNING No usable perl executable found

==============================================================================
vim.lsp: require("vim.lsp.health").check()

- LSP log level : WARN
- Log path: C:\Users\billionaire\AppData\Local\nvim-data\lsp.log
- Log size: 938 KB

vim.lsp: Active Clients ~
- omnisharp (id=1, root_dir=C:/Users/billionaire/Documents/Projects/csharpforfinance/KrayProject)

==============================================================================
vim.treesitter: require("vim.treesitter.health").check()

- Nvim runtime ABI version: 14
- OK Parser: bash       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\bash.so
- OK Parser: c          ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\c.so
- OK Parser: c_sharp    ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\c_sharp.so
- OK Parser: diff       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\diff.so
- OK Parser: graphql    ABI: 13, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\graphql.so
- OK Parser: hcl        ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\hcl.so
- OK Parser: html       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\html.so
- OK Parser: http       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\http.so
- OK Parser: javascript ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\javascript.so
- OK Parser: jsdoc      ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\jsdoc.so
- OK Parser: json       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\json.so
- OK Parser: jsonc      ABI: 13, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\jsonc.so
- OK Parser: lua        ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\lua.so
- OK Parser: luadoc     ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\luadoc.so
- OK Parser: luap       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\luap.so
- OK Parser: markdown   ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\markdown.so
- OK Parser: markdown_inline ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\markdown_inline.so
- OK Parser: printf     ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\printf.so
- OK Parser: python     ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\python.so
- OK Parser: query      ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\query.so
- OK Parser: regex      ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\regex.so
- OK Parser: sql        ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\sql.so
- OK Parser: terraform  ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\terraform.so
- OK Parser: toml       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\toml.so
- OK Parser: tsx        ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\tsx.so
- OK Parser: typescript ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\typescript.so
- OK Parser: vim        ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\vim.so
- OK Parser: vimdoc     ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\vimdoc.so
- OK Parser: xml        ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\xml.so
- OK Parser: yaml       ABI: 14, path: C:\Users\billionaire\AppData\Local\nvim-data\lazy\nvim-treesitter\parser\yaml.so
- OK Parser: c          ABI: 14, path: C:\tools\neovim\nvim-win64\lib\nvim\parser\c.dll
- OK Parser: lua        ABI: 14, path: C:\tools\neovim\nvim-win64\lib\nvim\parser\lua.dll
- OK Parser: query      ABI: 14, path: C:\tools\neovim\nvim-win64\lib\nvim\parser\query.dll
- OK Parser: vim        ABI: 14, path: C:\tools\neovim\nvim-win64\lib\nvim\parser\vim.dll
- OK Parser: vimdoc     ABI: 14, path: C:\tools\neovim\nvim-win64\lib\nvim\parser\vimdoc.dll

==============================================================================
which-key: require("which-key.health").check()

- OK Most of these checks are for informational purposes only.
  WARNINGS should be treated as a warning, and don't necessarily indicate a problem with your config.
  Please |DON't| report these warnings as an issue.

Checking your config ~
- OK |mini.icons| is installed
- OK |nvim-web-devicons| is installed

Checking for issues with your mappings ~
- OK No issues reported

checking for overlapping keymaps ~
- WARNING In mode `n`, <gc> overlaps with <gcc>, <gcO>, <gco>:
  - <gc>: Comment
  - <gcc>: Comment line
  - <gcO>: Add Comment Above
  - <gco>: Add Comment Below
- WARNING In mode `n`, <<Space>w> overlaps with <<Space>w|>, <<Space>wd>, <<Space>w->, <<C-W><Space>>, <<Space>ww>, <<Space>wm>:
  - <<Space>w>: windows
  - <<Space>w|>: Split Window Right
  - <<Space>wd>: Delete Window
  - <<Space>w->: Split Window Below
  - <<C-W><Space>>: Window Hydra Mode (which-key)
  - <<Space>ww>: Other Window
  - <<Space>wm>: Maximize Toggle
- WARNING In mode `x`, <a> overlaps with <al>, <a%>, <an>:
  - <a>: around
  - <al>: last
  - <an>: next
- WARNING In mode `x`, <i> overlaps with <il>, <in>:
  - <i>: inside
  - <il>: last
  - <in>: next
- WARNING In mode `o`, <a> overlaps with <al>, <an>:
  - <a>: around
  - <al>: last
  - <an>: next
- WARNING In mode `o`, <i> overlaps with <il>, <in>:
  - <i>: inside
  - <il>: last
  - <in>: next
- OK Overlapping keymaps are only reported for informational purposes.
  This doesn't necessarily mean there is a problem with your config.

Checking for duplicate mappings ~
- OK No duplicate mappings found

`

**Comment by folke** [comment]
You're not using a compatible Neovim version.
Not sure where you got it, but it's for sure messed up.
